pandas/core/reshape/merge.py

```diff
@@ -1639,3 +1634,16 @@ def _should_fill(lname, rname):
 
 def _any(x):
     return x is not None and com._any_not_none(*x)
+
+
+def validate_operand(obj):
+    if isinstance(obj, DataFrame):
+        return obj
+    elif isinstance(obj, Series):
+        if obj.name is None:
+            raise ValueError('Cannot merge a Series without a name')
+        else:
+            return obj.to_frame()
+    else:
```

Comment:
just raise, and more like

``Can only merge Series or DataFrame objects, a {obj} was passed``

Reply:
Thanks - done - updated the statement and made into a TypeError

Reply:
Is there a reason you changed this to TypeError? 
I would personally keep it as ValueError

Reply:
@jorisvandenbossche Thanks - I thought TypeError is more relevant for the statement below, as the message is telling the user "only objects of type Series or DataFrame can be merged"
Should I revert it back to ValueError?

Reply:
I don't think the one is necessarily much better than the other in this case, so I would personally rather keep it as it was to be conservative.

Reply:
Although eg `concat` raises TypeError if the input is not series or dataframe, so maybe TypeError is more consistent with pandas.

Reply:
Ok - not changing then and keeping it